src/python/dxpy/bindings/apollo/path_validator.py

```diff
@@ -57,12 +57,15 @@ def is_cohort_or_dataset(self):
         # since object is record:
         self.try_populate_record_http_request_info()
         if not (
-            ("Dataset" in self.record_http_request_info["recordTypes"])
-            or ("CohortBrowser" in self.record_http_request_info["recordTypes"])
+            ("Dataset" in self.record_http_request_info.get("recordTypes",[]))
```

Comment:
Minor, could be:

```
self.try_populate_record_http_request_info()

EXPECTED_TYPES = ["Dataset", "CohortBrowser"]

if any(x not in self.record_http_request_info["recordTypes"] and x not in self.record_http_request_info["types"] for x in EXPECTED_TYPES):
    self.throw_error(.......)
```

Reply:
Hi @JoeMulka I would agree with Ashkan this can get de-repeated slightly. I'd recommend establishing a method to retrieve types first and then use it with benefit to test certain types presence. Something along these lines perhaps?
```
def get_type(self):
    record_info = self.record_http_request_info # <-- we need to rename the field to the shorter version
    return set(record_info.get("record", []) + record_info.get("recordType", []));
```
And then:
```
if self.get_type() in ["Cohort", "Dataset"]:
   ...
```